feature_tests/src/selftype.rs

```diff
@@ -0,0 +1,20 @@
+#[diplomat::bridge]
+mod ffi {
+    struct RefList<'a> {
+        data: &'a i32,
+        next: Option<Box<Self>>,
+    }
+
+    impl<'b> RefList<'b> {
+        pub fn node(data: &'b i32) -> Self {
```

Comment:
thought: we actually should be banning references to anything other than slices, strings, and opaques

but that's for another PR, we can leave this in here